xtask/src/fpga.rs

```diff
@@ -231,18 +234,41 @@ pub(crate) fn fpga_entry(args: &Fpga) -> Result<()> {
                 rsync_file(&target_host, "all-fw.zip", false)?;
             }
         }
-        Fpga::BuildTest { target_host } => {
+        Fpga::BuildTest {
+            target_host,
+            caliptra_sw,
+        } => {
             println!("Building FPGA test");
             // Build test binaries in a docker container
             let home = std::env::var("HOME").unwrap();
             let project_root = PROJECT_ROOT.clone();
             let project_root = project_root.display();
 
             // TODO(clundin): Clean this docker command up.
-            Command::new("docker")
-                .current_dir(&*PROJECT_ROOT)
-                .args(["run", "--rm", &format!("-v{project_root}:/work-dir"), "-w/work-dir", &format!("-v{home}/.cargo/registry:/root/.cargo/registry"), &format!("-v{home}/.cargo/git:/root/.cargo/git"), "ghcr.io/chipsalliance/caliptra-build-image:latest", "/bin/bash", "-c", "(cd /work-dir && echo 'Cross compiling tests' && CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER=aarch64-linux-gnu-gcc cargo nextest archive --features=fpga_realtime --target=aarch64-unknown-linux-gnu --archive-file=caliptra-test-binaries.tar.zst --target-dir cross-target/ )"])
-                .status()?;
+            let mut cmd = Command::new("docker");
+            cmd.current_dir(&*PROJECT_ROOT).args([
+                "run",
+                "--rm",
+                &format!("-v{project_root}:/work-dir"),
+                "-w/work-dir",
+                &format!("-v{home}/.cargo/registry:/root/.cargo/registry"),
+                &format!("-v{home}/.cargo/git:/root/.cargo/git"),
+            ]);
+
+            // Add optional path to the caliptra-sw directory
+            if let Some(caliptra_sw) = caliptra_sw {
+                let basename = caliptra_sw.file_name().unwrap().to_str().unwrap();
+                cmd.arg(&format!(
+                    "-v{}:/{basename}",
+                    std::fs::canonicalize(&caliptra_sw)?.display()
+                ));
+            }
```

Comment:
If my local caliptra-sw is at `/home/clundin/code/sample` and I set `caliptra-sw` to `/home/clundin/code/sample`, will basename be `/home/clundin/code/sample`?

Reply:
No it would be `sample`. I mostly tested it with a relative path.

My path in the Cargo.toml was `../2.x`. I used `--caliptra-sw ../2.x` when running the command. So the argument became `-v/usr/local/google/home/zhalvorsen/caliptra/2.x:/2.x`.

Reply:
Hmm maybe we should add a note to the doc string that it has to be exact match for the path in cargo.toml